src/vivarium_gates_mncnh/components/intrapartum.py

```diff
@@ -0,0 +1,80 @@
+import pandas as pd
+from vivarium import Component
+from vivarium.framework.engine import Builder
+from vivarium.framework.event import Event
+from vivarium.framework.population import SimulantData
+
+from vivarium_gates_mncnh.constants.data_keys import NO_CPAP_INTERVENTION
+from vivarium_gates_mncnh.constants.data_values import (
+    COLUMNS,
+    CPAP_ACCESS_PROBABILITIES,
+    DELIVERY_FACILITY_TYPE_PROBABILITIES,
+    DELIVERY_FACILITY_TYPES,
+    SIMULATION_EVENT_NAMES,
+)
+from vivarium_gates_mncnh.utilities import get_location
+
+
+class Intrapartum(Component):
```

Comment:
If you put all of these decisions in one component, this component will get unmanageably large, since we will ultimately have a lot of interventions that are applied during the intrapartum period. You Should split this into a `FacilityChoice` component and a `CPAP` component. If you create the `CPAP` component such that it mirrors `vivarium_public_health.risk.Risk` closely enough, you should be able to replace your CPAP intervention component with a VPH `RiskEffect` component.